libs/core/src/service/document_service.rs

```diff
@@ -41,6 +65,27 @@ impl<Client: Requester> CoreState<Client> {
         let hmac = tree.find(&id)?.document_hmac();
         document_repo::insert(&self.config, &id, hmac, &encrypted_document)?;
 
+        let mut doc_events = self
+            .db
+            .doc_events
+            .data()
+            .get(&id)
+            .unwrap_or(&Vec::new())
+            .clone();
+
+        doc_events.sort_by(|a, b| b.cmp(a)); //sort in descending order
+        let latest_event = doc_events.iter().find(|e| matches!(e, DocEvents::Read(_)));
```

Comment:
i think you just need to peak at the last element and determine whether enough time has elapsed before writing

Reply:
there is also a copy paste related mistake here, avoid copy pasting things